pkg/login/social/socialimpl/service.go

```diff
@@ -222,7 +222,12 @@ func (ss *SocialService) getUsageStats(ctx context.Context) (map[string]any, err
 	return m, nil
 }
 
-func createOAuthConnector(name string, info *social.OAuthInfo, cfg *setting.Cfg, ssoSettings ssosettings.Service, features *featuremgmt.FeatureManager, cache remotecache.CacheStorage) (social.SocialConnector, error) {
+func createOAuthConnector(name string, settings map[string]any, cfg *setting.Cfg, ssoSettings ssosettings.Service, features *featuremgmt.FeatureManager, cache remotecache.CacheStorage) (social.SocialConnector, error) {
+	info, err := connectors.CreateOAuthInfoFromKeyValues(settings)
+	if err != nil {
+		return nil, err
```

Comment:
The `createOAuthConnector` should work differently when `ssoSettingApi` is enabled: it shouldn't error out but initialize an empty `info` and construct the provider and when the feature toggle is disabled then error out.